lib/modules/versioning/ruby/index.spec.ts

```diff
@@ -293,23 +313,66 @@ describe('modules/versioning/ruby/index', () => {
     ${'v1.0.3'}              | ${'replace'}         | ${'1.0.3'}     | ${'1.2.3'}   | ${'v1.2.3'}
     ${'= 1.0.3'}             | ${'replace'}         | ${'1.0.3'}     | ${'1.2.3'}   | ${'= 1.2.3'}
     ${'!= 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'!= 1.0.3'}
+    ${'!= 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.0.2'}   | ${'!= 1.0.3'}
+    ${'!= 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.0.3'}   | ${'!= 1.0.3'}
+    ${'> 1.0.3'}             | ${'replace'}         | ${'1.0.4'}     | ${'1.2.3'}   | ${'> 1.0.3'}
+    ${'> 1.2.3'}             | ${'replace'}         | ${'1.0.0'}     | ${'1.0.3'}   | ${'> 1.0.2'}
     ${'< 1.0.3'}             | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'< 1.2.4'}
+    ${'< 1.2.3'}             | ${'replace'}         | ${'1.0.0'}     | ${'1.0.3'}   | ${'< 1.2.3'}
     ${'< 1.2.2'}             | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'< 1.2.4'}
     ${'< 1.2.3'}             | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'< 1.2.4'}
     ${'< 1.2'}               | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'< 1.3'}
     ${'< 1'}                 | ${'replace'}         | ${'0.9.0'}     | ${'1.2.3'}   | ${'< 2'}
     ${'< 1.2.3'}             | ${'replace'}         | ${'1.0.0'}     | ${'1.2.2'}   | ${'< 1.2.3'}
     ${'>= 1.0.3'}            | ${'replace'}         | ${'1.0.3'}     | ${'1.2.3'}   | ${'>= 1.0.3'}
+    ${'>= 1.0.3'}            | ${'replace'}         | ${'1.0.3'}     | ${'1.0.2'}   | ${'>= 1.0.2'}
     ${'<= 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'<= 1.2.3'}
     ${'<= 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.0.2'}   | ${'<= 1.0.3'}
-    ${'~> 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'~> 1.2.0'}
+    ${'~> 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'~> 1.2.3'}
```

Comment:
This is a tough one. You could argue for `~> 1.2.0` too. What are your thoughts?

Reply:
When I wrote this, I interpreted the strategy as a conditional version of `bump`. But as you pointed out,

- npm's testcase says `0.5.6` should resolve as `^0.5.0`.
- it can be interpreted as a variation of `widen` that demands the use of the same notation.

Therefore I'm willing to change it back to `~> 1.2.0` here.

Reply:
Yes please do 1.2.0

Reply:
Fixed in https://github.com/renovatebot/renovate/pull/17091/commits/037ac4237e400d5d09ae51e9e5940e8f731ecc8e.

Reply:
please read the contribution guide
 https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#resolve-review-comments-instead-of-commenting